Firebase/Core/FIRApp.m

```diff
@@ -107,15 +108,22 @@ + (void)configure {
     // since the app isn't configured at all, so only rely on the Info.plist value.
     NSNumber *collectionEnabledPlistValue = [[self class] readDataCollectionSwitchFromPlist];
     if (collectionEnabledPlistValue == nil || [collectionEnabledPlistValue boolValue]) {
-      [[NSNotificationCenter defaultCenter]
-          postNotificationName:kFIRAppDiagnosticsNotification
-                        object:nil
-                      userInfo:@{
-                        kFIRAppDiagnosticsConfigurationTypeKey : @(FIRConfigTypeCore),
-                        kFIRAppDiagnosticsErrorKey : [FIRApp errorForMissingOptions]
-                      }];
+      if (NSClassFromString(@"FIRDiagnostics")) {
```

Comment:
I would move this check to a class in `DiagnosticsInterop` just to hide this detail from a client.

Reply:
Agreed - one of the main points of Interop is to use protocol-based calling instead of reflection.

Here and below.

Reply:
This sounds reasonable--but is it ok to have logic in an interop library? All of the other interop libraries only consist of headers.

Reply:
Interop pod's should only be protocol headers. the Implementations should be in a connector.

Reply:
The connector pattern is very hard to implement here in particular because of a few things:

- A static variable is needed to refer to the core diagnostics connector, but AFAICT, the first method to log is non-deterministic. This just makes things a bit messy, and we could @synchronize to ensure that setting that variable is static, it's not actually thread safe.
- In +configure, the notification that's sent out initialize diagnostics occurs before configuration of any app, even the default app. FIR_COMPONENT expects a container which is initialized with a FIRApp instance. This is a chicken and egg problem.

Reply:
Moved the logging logic to a separate connector class.